src/ResultSet.jl

```diff
@@ -41,45 +41,56 @@ end
 export columns
 
 # accessors 
-columns(rs::ResultSet) = rs.columns
-Base.names(rs::ResultSet) = rs.names
-Base.size(rs::ResultSet) = rs.size
-Base.size(rs::ResultSet, i::Integer) = rs.size[i]
+columns(rs::ResultSet) = getfield(rs, :columns)
+Base.names(rs::ResultSet) = getfield(rs, :names)
+
+Base.size(rs::ResultSet) = getfield(rs, :size)
+Base.size(rs::ResultSet, i::Integer) = getfield(rs, :size)[i]
```

Comment:
I think this 2nd `size` method and the `length` methods aren't needed if you implement the first `size` method, but that also might require `ResultSet` to be a subtype of `AbstractArray`. Note that I recently switched `CSV.File` to be `CSV.File <: AbstractVector{CSV.Row}` and it's made things a little more convenient in a couple of ways.